g2p/cli.py

```diff
@@ -677,3 +679,59 @@ def scan(lang, path):
         if unmapped:
             LOGGER.warning("The following characters are not mapped:")
             print(unmapped)
+
+
+@click.option(
+    "--all", "all_mappings", is_flag=True, help="Display all cached mappings."
+)
+@click.option("--csv", is_flag=True, help="Output mappings in CSV format.")
+@click.argument("lang2", required=False, default=None)
+@click.argument("lang1", required=False, default=None)
+@cli.command(context_settings=CONTEXT_SETTINGS, short_help="Show cached mappings.")
+def show_mappings(lang1, lang2, all_mappings, csv):
+    """Show cached mappings, as last updated by "g2p update".
+
+    Mappings on the path from LANG1 to LANG2 are displayed.
+    If only LANG1 is used, all mappings to or from LANG1 are displayed.
+    """
+
+    if all_mappings:
+        mappings = (
+            Mapping(in_lang=m["in_lang"], out_lang=m["out_lang"])
+            for m in MAPPINGS_AVAILABLE
+        )
+
+    elif lang1 is not None and lang2 is not None:
+        try:
+            transducer = make_g2p(lang1, lang2)
+        except (NoPath, InvalidLanguageCode) as e:
+            raise click.UsageError(
+                f'Cannot find mapping from "{lang1}" to "{lang2}": {e}'
+            ) from e
+
+        if isinstance(transducer, Transducer):
+            mappings = [transducer.mapping]
+        else:
+            mappings = (t.mapping for t in transducer._transducers)
+
+    elif lang1 is not None:
+        mappings = [
+            Mapping(in_lang=m["in_lang"], out_lang=m["out_lang"])
+            for m in MAPPINGS_AVAILABLE
+            if m["in_lang"] == lang1 or m["out_lang"] == lang1
+        ]
+        if not mappings:
+            raise click.BadParameter(
+                f'No mapping found to or from "{lang1}".', param_hint="lang1"
+            )
+
+    else:
```

Comment:
I think this is a bit cryptic - if a user types `g2p show-mappings` without the --all flag, it will just say "nothing to do!". This message should either tell the user to use the --all flag or (better yet) remove the all flag and print the total mappings in the else case.